configs/api/api.go

```diff
@@ -115,6 +119,47 @@ func (a *API) setConfig(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+func (a *API) validateAlertmanagerConfig(w http.ResponseWriter, r *http.Request) {
+	cfg, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Error reading request body: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err = validateAlertmanagerConfig(string(cfg)); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		util.WriteJSONResponse(w, map[string]string{
+			"status": "error",
+			"error":  err.Error(),
+		})
```

Comment:
I chose a similar JSON response format here as we use for validating expressions (https://github.com/weaveworks/cortex/blob/master/distributor/http_server.go#L72-L110), which is in turn similar to what the Prometheus query API generally returns. But happy to change it to anything else if desired. Last time, @davkal preferred this similar format.